### DISCUSSION TITLE
Demes integration

### DISCUSSION THREAD

**Discussion by sunyatin** [tskit-dev/msprime#1710] [Q&A] [answered]
Dear all!

Thank you *so* much for this tool. It is beyond cool. I had a question about integration with _Demes_: is it (currently in the near future if you think about implementing this option) possible to export a _msprime_ (legacy or >1.0) Demography to a _Demes_-formatted YAML file?

Thank you for your answer and best to you all :)

**Comment by jeromekelleher** [comment, marked as answer]
Thanks for the kind words @sunyatin!

Exporting to demes is certainly something I'd like to do, but not something I've sat down and thought about hard, so I'm not sure how easy it'll be. If you'd like to help with this I'd be more than happy to discuss it.

@apragsdale @grahamgower, have you any thoughts here?

**Reply by grahamgower** [reply]
There is/was a function to do this in the `demes` python package (for the legacy msprime API only). If you pip install `demes` (version 0.1 or 0.1.1), it can be accessed as `demes.convert.from_msprime()`. This is not in the `demes` documentation, and has actually been removed a few days ago (so won't be in demes version 0.1.2 when that is released this week). The reason for its removal, is that we expect to have this functionality in msprime instead.

If you have demes 0.1.1 installed, the help info for the converter function can be accessed like so:
```
$ python -c "import demes.convert; help(demes.convert.from_msprime)"

Help on function from_msprime in module demes.convert.msprime_:

from_msprime(population_configurations=None, demographic_events=None, migration_matrix=None, pop_names=None) -> demes.demes.Graph
    Convert an msprime demography into a demes graph.

    :param population_configurations: A list of population configurations.
    :type population_configurations: list[msprime.PopulationConfiguration]
    :param demographic_events: A list of demographic events.
    :type demographic_events: list[msprime.DemographicEvent]
    :param migration_matrix: The initial migration matrix.
    :type migration_matrix: list[list[float]]
    :param list[str] pop_names: A list of population names to use.
        If None, the names will be pop0, pop1, ..., popN.
    :return: A demes graph.
    :rtype: :class:`demes.Graph`
```

@sunyatin, hopefully that gives you the output you'd like. But please look at it critically, because this function was never rigorously tested!

To get this into msprime, we will need to update it to work with the msprime 1.0 demography API as well, and write bunch of tests.

**Reply by grahamgower** [reply]
Code here: https://github.com/popsim-consortium/demes-python/blob/0.1.1/demes/convert/msprime_.py#L177

**Reply by sunyatin** [reply]
AMAZING! Thank you very much Graham. For now, just wanted to do coarse visualization with DemesDraw using the YAML output, nothing very rigorous. But will anyway check the output and compare with a script that I also wrote to graphically represent _msprime_ Demography. Thank you very much again. Best to you two!
